File: Blind 75 LeetCode/Chahat Rajput/3 Sum/3_sum.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums)
{
    vector<vector<int>> ans;
    sort(nums.begin(),nums.end());
    int n=nums.size();
    for (int i = 0; i < n; i++)
    {
        int j = i + 1, k = n - 1;
        if (i > 0 && nums[i] == nums[i - 1]){continue;}
        while (j < k)
        {
            int sum = nums[i] + nums[j] + nums[k];
            if (sum < 0)
            {
                j++;
            }
            else if (sum > 0)
            {
                k--;
            }
            else if (sum == 0)
            {
                vector<int> temp = {nums[i], nums[j], nums[k]};
                ans.push_back(temp);

                j++;
                k--;

                while(j<k && nums[j] == nums[j-1]){j++;}
                while(j<k && nums[k] == nums[k+1]){k--;}
            }
        }
    }
    return ans;
}
};

File: Blind 75 LeetCode/Subhangi dhasmana/3 sum/3 sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ans = []
        nums.sort()
        for i in range(len(nums)):
            if i>0 and nums[i]==nums[i-1]: continue
            j=i+1
            k=len(nums)-1
            while(j<k):
                totSum = nums[i]+nums[j]+nums[k]
                if totSum<0:
                    j+=1
                elif totSum>0:
                    k-=1
                else:
                    temp = [nums[i], nums[j], nums[k]]
                    ans.append(temp)
                    j+=1
                    k-=1
                    while nums[j]==nums[j-1] and j<k:
                        j+=1
                    while nums[k]==nums[k+1] and j<k:
                        k-=1
        return ans